dietpi/misc/dietpi-nordvpn

```diff
@@ -172,76 +166,81 @@ _EOF_
 			if [[ -f /sys/class/net/$NET_DEV/statistics/rx_bytes && -f /sys/class/net/$NET_DEV/statistics/tx_bytes ]]; then
 
 				net_rx_byte=$(</sys/class/net/$NET_DEV/statistics/rx_bytes)
-				net_tx_byte=$(</sys/class/net/$NET_DEV/statistics/tx_bytes)
 				net_rx_mb='Unknown'
-				if disable_error=1 G_CHECK_VALIDINT $net_rx_byte && (( $net_rx_byte > 0 )); then
-
-					net_rx_mb="$(( $net_rx_byte / 1024 / 1024 ))MB"
-
-				fi
+				disable_error=1 G_CHECK_VALIDINT "$net_rx_byte" 1 && net_rx_mb="$(( $net_rx_byte / 1024 / 1024 ))MB"
 
+				net_tx_byte=$(</sys/class/net/$NET_DEV/statistics/tx_bytes)
 				net_tx_mb='Unknown'
-				if disable_error=1 G_CHECK_VALIDINT $net_tx_byte && (( $net_tx_byte > 0 )); then
-
-					net_tx_mb="$(( $net_tx_byte / 1024 / 1024 ))MB"
-
-				fi
+				disable_error=1 G_CHECK_VALIDINT "$net_tx_byte" 1 && net_tx_mb="$(( $net_tx_byte / 1024 / 1024 ))MB"
 
 			fi
 
 			text_status+=" - Usage     : Sent = $net_tx_mb | Recieved = $net_rx_mb"
 
 		else
 
-			text_status+=' - Disconnected\n'
-			text_status+=' - NordVPN subscription: https://go.nordvpn.net/aff_c?offer_id=15&aff_id=5305&url_id=902'
+			(( $NORDVPN_SERVICE )) && text_status+=' - Disconnected' || text_status+=' - Not configured'
+			text_status+='\n - NordVPN subscription: https://go.nordvpn.net/aff_c?offer_id=15&aff_id=5305&url_id=902'
 
 		fi
 
 		G_WHIP_MENU_ARRAY=('' '●─ Global Options ')
 		G_WHIP_MENU_ARRAY+=('Username' ": [$NORDVPN_USERNAME]")
-		G_WHIP_MENU_ARRAY+=('Password' ": [$NORDVPN_PASSWORD]")
+		G_WHIP_MENU_ARRAY+=('Password' ": [${NORDVPN_PASSWORD//?/*}]")
 		G_WHIP_MENU_ARRAY+=('Server' ": [$NORDVPN_SERVER]")
-		G_WHIP_MENU_ARRAY+=('Refresh' ': Update VPN connection status')
-		(( $NORDVPN_CONNECTED )) && G_WHIP_MENU_ARRAY+=('Disconnect' '')
+		if (( $NORDVPN_SERVICE )); then
+
+			local autostart_enabled=0 autostart_text='Off'
+			systemctl -q is-enabled dietpi-nordvpn && { autostart_enabled=1 autostart_text='On'; }
+			G_WHIP_MENU_ARRAY+=('Auto start' ": [$autostart_text]")
+
+			G_WHIP_MENU_ARRAY+=('Refresh' ': Update VPN connection status')
+			(( $NORDVPN_CONNECTED )) && G_WHIP_MENU_ARRAY+=('Disconnect' '')
+
+		fi
 		G_WHIP_MENU_ARRAY+=('' '●─ Save Settings ')
 		G_WHIP_MENU_ARRAY+=('Apply' ': Save settings and restart VPN connection')
 
 		G_WHIP_DEFAULT_ITEM=$LAST_SELECTED_NAME
 		G_WHIP_BUTTON_CANCEL_TEXT='Exit'
 		if G_WHIP_MENU "$text_status"; then
 
-			LAST_SELECTED_NAME="$G_WHIP_RETURNED_VALUE"
+			LAST_SELECTED_NAME=$G_WHIP_RETURNED_VALUE
 
 			if [[ $G_WHIP_RETURNED_VALUE == 'Apply' ]]; then
 
 				Save_Settings
 
-			elif [[ $G_WHIP_RETURNED_VALUE == 'Username' ]]; then
+			elif [[ $G_WHIP_RETURNED_VALUE == 'Auto start' ]]; then
 
-				G_WHIP_DEFAULT_ITEM=$NORDVPN_USERNAME
-				if G_WHIP_INPUTBOX 'Please enter your NordVPN username:'; then
+				if (( $autostart_enabled )); then
+
+					G_RUN_CMD systemctl disable dietpi-nordvpn
+
+				else
 
-					NORDVPN_USERNAME=$G_WHIP_RETURNED_VALUE
+					G_RUN_CMD systemctl enable dietpi-nordvpn
```

Comment:
@Fourdee 
I think when enabling the service it makes sense to apply current chosen settings as well? Otherwise e.g. user chooses some settings, then enables the service, then exits, then the service does not contain the chosen values. Of course one should "Apply" always, but we want to be failsafe here? Or are there any cases where you want to enable the service without having the currently chosen/entered (thus visible in whiptail menu) values applied for this?